php/Block.php

```diff
@@ -48,81 +48,130 @@ public function init() {
-                $post_count = count(
-                    get_posts(
-						[
-							'post_type' => $post_type_slug,
-							'posts_per_page' => -1,
-						]
-					)
-                );
-
-				?>
-				<li><?php echo 'There are ' . $post_count . ' ' .
-					  $post_type_object->labels->name . '.'; ?></li>
-			<?php endforeach;	?>
-			</ul><p><?php echo 'The current post ID is ' . $_GET['post_id'] . '.'; ?></p>
-
-			<?php
-			$query = new WP_Query(  array(
-				'post_type' => ['post', 'page'],
-				'post_status' => 'any',
-				'date_query' => array(
-					array(
-						'hour'      => 9,
-						'compare'   => '>=',
-					),
-					array(
-						'hour' => 17,
-						'compare'=> '<=',
-					),
+	public function render_callback( $attributes ) {
+
+		$class_name = 'site-counts-container';
+		if ( isset( $attributes['className'] ) ) {
+			$class_name .= ' ' . $attributes['className'];
+		}
+		$current_post = get_the_ID();
+		$html         = array();
+
+		// Open container.
+		$html[] = '<div class="' . esc_attr( $class_name ) . '">';
+
+		// Get the counts.
+		$html[] = $this->get_count_posts_html();
+
+		// translators: placeholder is the current post ID.
+		$html[] = '<p>' . sprintf( __( 'The current post ID is %s', 'site-counts' ), $current_post ) . '</p>';
+
+		// Get the 5 latest posts.
+		$html[] = $this->get_latest_posts_html( $current_post );
+
+		// Close container.
+		$html[] = '</div>';
+
+		$html_string = implode( $html );
+
+		return wp_kses_post( $html_string );
+	}
+
+	/**
+	 * Get the Post type counts.
+	 *
+	 * @return string
+	 */
+	protected function get_count_posts_html() {
+
+		$html       = array();
+		$html[]     = '<h2>' . esc_html__( 'Post Counts', 'site-counts' ) . '</h2>';
+		$html[]     = '<ul>';
+		$post_types = get_post_types( array( 'public' => true ) );
+		foreach ( $post_types as $post_type_slug ) {
+			$post_type_object = get_post_type_object( $post_type_slug );
+			$counts           = wp_count_posts( $post_type_slug );
+			$post_count       = 'attachment' === $post_type_slug ? (int) $counts->inherit : (int) $counts->publish;
+			$single_label     = $post_type_object->labels->singular_name;
+			$plural_label     = $post_type_object->labels->name;
+			$note             = sprintf(
+			// translators: Post count difference. %1$d is number of posts, %2$s & %3$s is the post type singular and plural names.
+				_n( 'There is %1$d %2$s', 'There are %1$d %3$s', $post_count, 'site-counts' ), // phpcs:ignore WordPress.WP.I18n.MismatchedPlaceholders
+				$post_count,
+				$single_label,
+				$plural_label
+			);
+
+			// Add to the HTML.
+			$html[] = '<li>' . $note . '</li>';
+		}
+
+		$html[] = '</ul>';
+
+		return implode( $html );
+	}
+
+	/**
+	 * Get 5 of the latest posts.
+	 *
+	 * @param int $exclude The ID to exclude.
+	 *
+	 * @return string
+	 */
+	protected function get_latest_posts_html( $exclude ) {
+
+		$html  = array();
+		$args  = array(
+			'post_type'              => array( 'post', 'page' ),
```

Comment:
Since we have hard coded taxonomies in the query, we could remove `page` since the query would never get any pages. I left in however as I don't have context of the project. i.e It could have a programmatically added `tag` or `category` as part of an internal workflow.